src/include/simeng/OS/SyscallHandler.hh

```diff
@@ -212,22 +334,37 @@ class SyscallHandler {
   int64_t writev(int64_t fd, const void* iovdata, int iovcnt);
 
  private:
-  /** Pointer reference to SimOS object. */
-  SimOS* os_ = nullptr;
-
-  /** Returns the correct Dirfd depending on the pathname and dirfd given to
+  /** Returns the correct dirFd depending on the pathname and dirFd given to
    * syscall. */
   uint64_t getDirFd(int64_t dfd, std::string pathname);
 
   /** If the given filepath points to a special file, the filepath is replaced
    * to point to the SimEng equivalent. */
   std::string getSpecialFile(const std::string filename);
 
+  /** Pointer reference to SimOS object. */
+  SimOS* OS_ = nullptr;
+
+  /** A shared pointer to the simulation memory. */
+  std::shared_ptr<simeng::memory::Mem> memory_;
+
+  /** A queue to hold all outstanding syscalls. */
+  std::queue<SyscallInfo> syscallQueue_;
```

Comment:
Me and @FinnWilkinson  looked at it, std::list seems to be a better substitute it has the same functionality, it can pop and push both and head and tail, and return an iterator i.e enabled insertions and removals from any index in the list. All of this with a much smaller memory footprint than std::queue. So change at your own discretion

Reply:
Whilst I agree, I think this can be changed into a future PR (such as your `futex` one)

Reply:
Leaving to future PR for reassessment